src/chips/poseidon/poseidon_spec.rs

```diff
@@ -1,23 +1,23 @@
 //! This file was generated by running generate_params.py
-//! Specification for rate 3 Poseidon using the BN256 curve.
+//! Specification for rate 6 Poseidon using the BN256 curve.
 //! Patterned after [halo2_gadgets::poseidon::primitives::P128Pow5T3]
-use crate::chips::poseidon::poseidon_params_leaf;
+use crate::chips::poseidon::poseidon_params;
 use halo2_gadgets::poseidon::primitives::*;
 use halo2_proofs::arithmetic::Field;
 use halo2_proofs::halo2curves::bn256::Fr as Fp;
 
 #[derive(Debug, Clone, Copy)]
-pub struct MySpec;
+pub struct PoseidonSpec;
```

Comment:
Even though we had this bug before, I think that the idea that Han suggested to just change the L while keeping the same specs for the Poseidon is still worth exploring. If we manage to use a single poseidon spec for any N_ASSETS that would simplify things a lot

Reply:
Ok, I'm not understanding it then. Why would you even need to generate different parameters if this is independent of the N_ASSETS

Reply:
Removed the reference to the number of assets.